lib/hooks.ts

```diff
@@ -30,13 +30,17 @@ export function useProducts() {
     };
 }
 
-export function useProductList() {
+export function useProductList(page?: number, limit?: number) {
     const encodedContext = useSession()?.context;
+    const pageQuery = page ? `&page=${page}` : '';
+    const limitQuery = limit ? `&limit=${limit}` : '';
+    const query = `${pageQuery}${limitQuery}`;
     // Use an array to send multiple arguments to fetcher
-    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext] : null, fetcher);
+    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext, query] : null, fetcher);
 
     return {
-        list: data,
+        list: data ? data.data : [],
+        meta: data ? data.meta : {},
```

Comment:
I wouldn't initialize this to an empty object - this can have unintended side-effects. We can simply do `data?.meta` here. Note: in this instance we should use optional chaining on meta since `data` is in fact a destructured (nested) object - https://github.com/bigcommerce/sample-app-nodejs/blob/main/pages/api/products/list.ts#L13
```suggestion
        meta: data?.meta,
```